### PR TITLE
Change unique_together to use constraints, replace use of index_together

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: dash/categories/models.py [hunk 1]
```diff
@@ -50,7 +50,9 @@ def __str__(self):
 
     class Meta:
         ordering = ["name"]
-        unique_together = ("name", "org")
+        constraints = [
+            models.UniqueConstraint(fields=["name", "org"], name="categories_category_name_318a92307e6f39fb_uniq")
```
[the remark this answers is not in the document]
Comment (norkans7) [reply dash/categories/models.py:54]: I reused the existing index name

[the remark this answers is not in the document]
Comment (rowanseymour) [reply dash/categories/models.py]: I get that but I'm not sure it's worth it or that that name is guaranteed to be the one used on every deployment

File: dash/dashblocks/migrations/0011_alter_dashblock_index_together_and_more.py [hunk 2]
```diff
@@ -10,12 +10,13 @@ class Migration(migrations.Migration):
...
-            index_together={("org", "is_active", "dashblock_type", "priority")},
+        migrations.AddIndex(
+            model_name="dashblock",
+            index=models.Index(
+                fields=["org", "is_active", "dashblock_type", "priority"], name="dashblocks__org_id_024805_idx"
+            ),
         ),
-        migrations.AlterIndexTogether(
-            name="dashblocktype",
-            index_together={("slug", "name")},
```
Comment (rowanseymour) [inline dash/dashblocks/migrations/0011_alter_dashblock_index_together_and_more.py:19]: Is there a reason to rewrite existing migrations? I think we just let it generate new ones which I think will noop anyway.

File: dash/orgs/migrations/0031_alter_orgbackend_index_together.py [hunk 3]
```diff
@@ -9,8 +9,8 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        migrations.AlterIndexTogether(
-            name="orgbackend",
-            index_together={("org", "is_active", "slug")},
-        ),
+        migrations.AddIndex(
+            model_name="orgbackend",
+            index=models.Index(fields=["org", "is_active", "slug"], name="orgs_orgbac_org_id_607508_idx"),
```
Comment (rowanseymour) [inline dash/orgs/migrations/0031_alter_orgbackend_index_together.py:14]: I don't think there's guarantee that generated index names are the same across deploys... just let Django make a new migrations

Comment (rowanseymour) [reply dash/orgs/migrations/0031_alter_orgbackend_index_together.py:14]: Or at least get Dash running on 5.0, which I think is smarter about changes to `index_together` than 4.2

Comment (norkans7) [reply dash/orgs/migrations/0031_alter_orgbackend_index_together.py:14]: Yes, I am going to remove the update to 5.1 and change the index_together changes first

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**